presto-accumulo/src/main/java/com/facebook/presto/accumulo/model/RowSchema.java

```diff
@@ -0,0 +1,175 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.accumulo.model;
+
+import com.facebook.presto.spi.PrestoException;
+import com.facebook.presto.spi.type.Type;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import static com.facebook.presto.accumulo.AccumuloErrorCode.COLUMN_NOT_FOUND;
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.lang.String.format;
+
+/**
+ * Class to define the schema of a Row, stored as a list of {@link AccumuloColumnHandle}.
+ */
+public class RowSchema
+{
+    private List<AccumuloColumnHandle> columns = new ArrayList<>();
+
+    /**
+     * Gets a new instance of RowSchema.
+     *
+     * @return A new RowSchema
+     */
+    public static RowSchema newRowSchema()
+    {
+        return new RowSchema();
+    }
+
+    /**
+     * Adds the Row ID column to the schema
+     *
+     * @param name Name of the row ID column
+     * @param type Presto type of the column
+     * @return this, for chaining
+     */
+    public RowSchema addRowId(String name, Type type)
+    {
+        columns.add(new AccumuloColumnHandle(name, null, null, type, columns.size(),
+                "Accumulo row ID", false));
+        return this;
+    }
+
+    /**
+     * Appends a new non-indexed column to the end of the schema.
+     *
+     * @param prestoName Presto column name
+     * @param family Accumulo column family, optional if column is row ID
+     * @param qualifier Accumulo column qualifier, optional if column is row ID
+     * @param type Presto type of the column
+     * @return this, for schema
+     */
+    public RowSchema addColumn(String prestoName, Optional<String> family, Optional<String> qualifier, Type type)
+    {
+        return addColumn(prestoName, family, qualifier, type, false);
+    }
+
+    /**
+     * Appends a new column to the end of the schema.
+     *
+     * @param prestoName Presto column name
+     * @param family Accumulo column family, optional if column is row ID
+     * @param qualifier Accumulo column qualifier, optional if column is row ID
+     * @param type Presto type of the column
+     * @param indexed True if indexed, false otherwise
+     * @return this, for schema
+     */
+    public RowSchema addColumn(String prestoName, Optional<String> family, Optional<String> qualifier, Type type,
+            boolean indexed)
+    {
+        columns.add(new AccumuloColumnHandle(prestoName, family, qualifier, type,
+                columns.size(),
+                format("Accumulo column %s:%s. Indexed: %b", family, qualifier, indexed),
```

Comment:
The column comments are for informational purposes only when a user describes the table.